pkg/whitesource/whitesource.go

```diff
@@ -186,56 +172,33 @@ func (s *System) GetProjectVitals(projectToken string) (*Project, error) {
 		ProjectToken: projectToken,
 	}
 
-	respBody, err := s.sendRequest(req)
+	err := s.sendRequestAndDecodeJSON(req, &wsResponse)
 	if err != nil {
-		return nil, errors.Wrap(err, "WhiteSource request failed")
+		return Project{}, errors.Wrap(err, "WhiteSource request failed")
 	}
 
-	err = json.Unmarshal(respBody, &wsResponse)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to parse WhiteSource response")
+	if len(wsResponse.ProjectVitals) == 0 {
+		return Project{}, errors.Wrapf(err, "no project with token '%s' found in WhiteSource", projectToken)
 	}
 
-	return &wsResponse.ProjectVitals[0], nil
+	return wsResponse.ProjectVitals[0], nil
 }
 
-// GetProjectByName returns the finds and returns a project by name
-func (s *System) GetProjectByName(productToken, projectName string) (*Project, error) {
-	var project *Project
+// GetProjectByName fetches all projects and returns the one matching the given projectName, or none, if not found
+func (s *System) GetProjectByName(productToken, projectName string) (Project, error) {
 	projects, err := s.GetProjectsMetaInfo(productToken)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to retrieve WhiteSource project meta info")
-	}
-
-	for _, proj := range projects {
-		if projectName == proj.Name {
-			project = &proj
-			break
-		}
-	}
-
-	// returns a nil project and no error if no project exists with projectName
-	return project, nil
-}
-
-// GetProjectsByIDs: get all project tokens given a list of project ids
-func (s *System) GetProjectsByIDs(productToken string, projectIDs []int64) ([]Project, error) {
```

Comment:
why removing this? We should keep this. This will be a re-use library for other steps using WhiteSource functionality

Reply:
I restored this.